app/src/contacts/contactsService.js

```diff
@@ -0,0 +1,76 @@
+(function() {
+    'use strict';
+
+    angular
+            .module('openeApp.contacts')
+            .factory('contactsService', contactsService);
+
+    contactsService.$inject = ['$http', '$q'];
+
+    function contactsService($http, $q) {
+        var service = {
+            //organizations
+            getOrganizations: getOrganizations,
+            getOrganization: getContact,
+            updateOrganization: updateContact,
+            createOrganization: createOrganization,
+            //persons
+            getPersons: getPersons,
+            getPerson: getContact,
+            updatePerson: updateContact,
+            createPerson: createPerson
+        };
+        return service;
+        
+        //organizations
+        function getOrganizations(searchTerm) {
+            return getContacts(searchTerm, 'contact:organization');
+        }
+        
+        function createOrganization(organization) {
+            return createContact(organization, 'ORGANIZATION');
+        }
+        
+        //persons
+        function getPersons(searchTerm) {
+            return getContacts(searchTerm, 'contact:person');
+        }
+        
+        function createPerson(organization) {
+            return createContact(organization, 'PERSON');
+        }
+        
+        //contacts
+        function getContacts(searchTerm, baseType) {
+            return $http.get('/alfresco/service/api/openesdh/contactsearch',
+                    {params: {baseType: baseType, term: searchTerm}})
+                    .then(successOrReject);
+        }
+
+        function getContact(storeProtocol, storeIdentifier, uuid) {
```

Comment:
It's just an opinion but probably it would be nicer to use nodeRef as a parameter and use alfrescoNodeUtils.processNodeRef() to constract the urls.
In the controller the nodeRef could be simply constructed of storeProtocol + '://' + storeType + '/' + id